src/server/pfs/pfs.go

```diff
@@ -1,14 +1,51 @@
 package pfs
 
 import (
-	"errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
 
 	"github.com/pachyderm/pachyderm/src/client/pfs"
 )
 
-var ErrFileNotFound = errors.New("file not found")
+type ErrFileNotFound struct {
+	error
+}
+
+type ErrRepoNotFound struct {
+	error
+}
+
+type ErrCommitNotFound struct {
+	error
+}
+
+type ErrParentCommitNotFound struct {
+	error
+}
 
-var ErrRepoNotFound = errors.New("repo not found")
+func NewErrFileNotFound(file string, repo string, commitID string) *ErrFileNotFound {
+	return &ErrFileNotFound{
+		error: grpc.Errorf(codes.NotFound, "File %v not found in repo %v at commit %v", file, repo, commitID),
+	}
+}
+
+func NewErrRepoNotFound(repo string) *ErrRepoNotFound {
+	return &ErrRepoNotFound{
+		error: grpc.Errorf(codes.NotFound, "repo %v not found", repo),
+	}
+}
+
+func NewErrCommitNotFound(repo string, commitID string) *ErrCommitNotFound {
+	return &ErrCommitNotFound{
+		error: grpc.Errorf(codes.NotFound, "commit %v not found in repo %v", commitID, repo),
+	}
+}
+
+func NewErrParentCommitNotFound(repo string, commitID string) *ErrParentCommitNotFound {
+	return &ErrParentCommitNotFound{
+		error: grpc.Errorf(codes.NotFound, "parent commit %v not found in repo %v", commitID, repo),
+	}
+}
 
```

Comment:
Will these errors get returned by the golang API? If so does it maybe make sense to put them in `src/client` so people can use them there?